File: spec/code_samples/Ruby/api@v1@contacts@lists@{listId}@contacts/patch.rb
require 'net/http'
require 'json'
require 'base64'
require 'uri'

api_key = 'YOUR_API_KEY'
api_secret = 'YOUR_API_SECRET'
api_host = 'YOUR_API_HOST' # Set YOUR_API_HOST to the regional host from the servers section in the docs
list_id = "YOUR_LIST_ID"

body = {
  "contactsToAddIds" => [
    "025e93d3-051b-43f9-b12e-4b5842228dee"
  ],
  "contactsToRemoveIds" => [
    "025e93d3-051b-43f9-b12e-4b5842228dee"
  ]
}

# HMAC authentication is also supported instead of Basic
uri = URI.parse("#{api_host}/api/v1/contacts/lists/#{list_id}/contacts")
http = Net::HTTP.new(uri.host, uri.port)
http.use_ssl = uri.scheme == 'https'

request = Net::HTTP::Patch.new(uri)
request['Authorization'] = 'Basic ' + Base64.strict_encode64("#{api_key}:#{api_secret}")
request['Accept'] = 'application/json'
request['Content-Type'] = 'application/json'
request.body = JSON.generate(body)

response = http.request(request)
puts response.code
puts response.body

File: spec/code_samples/PHP/api@v1@contacts@lists@{listId}@contacts/patch.php
<?php
$apiKey = 'YOUR_API_KEY';
$apiSecret = 'YOUR_API_SECRET';
$apiHost = 'YOUR_API_HOST'; // Set YOUR_API_HOST to the regional host from the servers section in the docs
$listId = "YOUR_LIST_ID";

$body = [
  'contactsToAddIds' => [
    '025e93d3-051b-43f9-b12e-4b5842228dee'
  ],
  'contactsToRemoveIds' => [
    '025e93d3-051b-43f9-b12e-4b5842228dee'
  ]
];

// HMAC authentication is also supported instead of Basic
$url = $apiHost . '/api/v1/contacts/lists/' . $listId . '/contacts';
$headers = [
    'Authorization: Basic ' . base64_encode($apiKey . ':' . $apiSecret),
    'Accept: application/json',
    'Content-Type: application/json',
];

$ch = curl_init($url);
curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'PATCH');
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($body));

$response = curl_exec($ch);
$status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
curl_close($ch);

echo $status, PHP_EOL;
echo $response, PHP_EOL;
